test/helpers/name.go

```diff
@@ -52,7 +53,14 @@ func ObjectPrefixForTest(t named) string {
 
 // ObjectNameForTest generates a random object name based on the test name.
 func ObjectNameForTest(t named) string {
-	return kmeta.ChildName(ObjectPrefixForTest(t), string(sep)+RandomString())
+	prefix := ObjectPrefixForTest(t)
+	suffix := string(sep) + RandomString()
+	limit := nameLengthLimit - len(suffix)
+	if len(prefix) < limit {
+		limit = len(prefix)
+	}
+
+	return kmeta.ChildName(prefix[:limit], suffix)
```

Comment:
does ChildName actually do anything here? IIUC ChildName truncates the name iff the prefix plus the suffix are bigger than 63 characters, but in this case we've already limited the prefix to `50 - the length of the suffix`, so - I think - the input to this method is always less than 50 characters

Reply:
Good caught. Removed.